src/sentry/preprod/api/endpoints/project_preprod_build_details.py

```diff
@@ -61,6 +66,41 @@ def get(self, request: Request, project, artifact_id) -> Response:
         except PreprodArtifact.DoesNotExist:
             return Response({"error": f"Preprod artifact {artifact_id} not found"}, status=404)
 
+        try:
+            size_metrics_qs = PreprodArtifactSizeMetrics.objects.select_related(
+                "preprod_artifact"
+            ).filter(
+                preprod_artifact__project=project,
+                preprod_artifact__id=artifact_id,
+            )
+            size_metrics_count = size_metrics_qs.count()
+            if size_metrics_count == 0:
+                logger.info("No size analysis results found for preprod artifact %s", artifact_id)
+                size_info = None
+            elif size_metrics_count > 1:
```

Comment:
This is a valid case, we should be querying for metrics where `metrics_artifact_type=MAIN_ARTIFACT`

Reply:
+1